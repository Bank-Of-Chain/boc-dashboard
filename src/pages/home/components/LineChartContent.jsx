import React from 'react'
import classNames from 'classnames'
import { Card, Button, Tabs, Tooltip } from 'antd'
import { LineEchart } from '@/components/echarts'
import { useDeviceType, DEVICE_TYPE } from '@/components/Container/Container'
import styles from '../style.less'

const { TabPane } = Tabs

export default function LineChartContent({
  isUsdi,
  loading = false,
  calDateRange = 7,
  onCalDateRangeClick = () => {},
  apyEchartOpt = {},
  tvlEchartOpt = {}
}) {
  const deviceType = useDeviceType()

  const chartResponsiveConfig = {
    [DEVICE_TYPE.Desktop]: {
      chartWrapperClassName: styles.chartDiv
    },
    [DEVICE_TYPE.Tablet]: {
      buttonProps: {
        size: 'small',
        style: { fontSize: '0.5rem' }
      },
      chartWrapperClassName: styles.chartDivMobile
    },
    [DEVICE_TYPE.Mobile]: {
      buttonProps: {
        size: 'small',
        style: { fontSize: '0.5rem' }
      },
      chartWrapperClassName: styles.chartDivMobile,
      tabClassName: styles.tabMobile
    }
  }[deviceType]

  return (
    <Card
      loading={loading}
      bordered={false}
      style={{ marginTop: 40 }}
    >
      <div className={styles.vaultKeyCard}>
        <Tabs
          animated
          className={classNames(chartResponsiveConfig.tabClassName)}
          tabBarExtraContent={
            <div className={styles.buttons}>
              <Tooltip title='last 7 days'>
                <Button
                  ghost
                  type={calDateRange === 7 ? 'primary' : ''}
                  onClick={() => onCalDateRangeClick(7)}
                  {...chartResponsiveConfig.buttonProps}
                >
                  WEEK
                </Button>
              </Tooltip>
              <Tooltip title='last 30 days'>
                <Button
                  ghost
                  type={calDateRange === 31 ? 'primary' : ''}
                  onClick={() => onCalDateRangeClick(31)}
                  {...chartResponsiveConfig.buttonProps}
                >
                  MONTH
                </Button>
              </Tooltip>
              <Tooltip title='last 365 days'>
                <Button
                  ghost
                  type={calDateRange === 365 ? 'primary' : ''}
                  onClick={() => onCalDateRangeClick(365)}
                  {...chartResponsiveConfig.buttonProps}
                >
                  YEAR
                </Button>
              </Tooltip>
            </div>
          }
        >
          <TabPane tab="APY (%)" key="apy">
            <div className={chartResponsiveConfig.chartWrapperClassName}>
              <LineEchart option={apyEchartOpt} style={{height: '100%', width: '100%'}}/>
            </div>
          </TabPane>
          <TabPane tab={isUsdi ? 'Total USDi Supply' : 'Total ETHi Supply'} key='totalSupply'>
            <div className={chartResponsiveConfig.chartWrapperClassName}>
              <LineEchart option={tvlEchartOpt} style={{ height: '100%', width: '100%' }} />
            </div>
          </TabPane>
        </Tabs>
      </div>
    </Card>
  )
}
