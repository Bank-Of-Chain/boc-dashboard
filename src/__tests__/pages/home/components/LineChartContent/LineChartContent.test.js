import React from 'react'
import { render } from '@testing-library/react'
import LineChartContent from '@/pages/home/components/LineChartContent'

test('LineChartContent Component Render', () => {
  const { asFragment } = render(
    <LineChartContent loading={false} calDateRange={() => {}} onCalDateRangeClick={() => {}} apyEchartOpt={{}} tvlEchartOpt={{}} />
  )
  expect(asFragment()).toMatchSnapshot()
})
