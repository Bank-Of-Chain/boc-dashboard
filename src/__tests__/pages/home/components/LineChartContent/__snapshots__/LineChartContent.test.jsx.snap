// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LineChartContent Component Render 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-small b-rd-5 css-dev-only-do-not-override-yp8pcc"
  >
    <div
      class="ant-card-body"
    >
      <div>
        <div
          class="ant-tabs ant-tabs-top css-dev-only-do-not-override-yp8pcc"
        >
          <div
            class="ant-tabs-nav"
            role="tablist"
          >
            <div
              class="ant-tabs-nav-wrap"
            >
              <div
                class="ant-tabs-nav-list"
                style="transform: translate(0px, 0px);"
              >
                <div
                  class="ant-tabs-tab ant-tabs-tab-active"
                  data-node-key="apy"
                >
                  <div
                    aria-controls="rc-tabs-test-panel-apy"
                    aria-selected="true"
                    class="ant-tabs-tab-btn"
                    id="rc-tabs-test-tab-apy"
                    role="tab"
                    tabindex="0"
                  >
                    APY (%)
                  </div>
                </div>
                <div
                  class="ant-tabs-tab"
                  data-node-key="totalSupply"
                >
                  <div
                    aria-controls="rc-tabs-test-panel-totalSupply"
                    aria-selected="false"
                    class="ant-tabs-tab-btn"
                    id="rc-tabs-test-tab-totalSupply"
                    role="tab"
                    tabindex="0"
                  >
                    Total Supply
                  </div>
                </div>
                <div
                  class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                />
              </div>
            </div>
            <div
              class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
            >
              <button
                aria-controls="rc-tabs-test-more-popup"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-hidden="true"
                class="ant-tabs-nav-more"
                id="rc-tabs-test-more"
                style="visibility: hidden; order: 1;"
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="ellipsis"
                  class="anticon anticon-ellipsis"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="ellipsis"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="ant-tabs-extra-content"
            >
              <div
                class="ant-select !b-rd-0 text-violet-400 css-dev-only-do-not-override-yp8pcc ant-select-single ant-select-show-arrow"
                style="width: 120px;"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                  />
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="ant-tabs-content-holder"
          >
            <div
              class="ant-tabs-content ant-tabs-content-top"
            >
              <div
                aria-hidden="false"
                aria-labelledby="rc-tabs-test-tab-apy"
                class="ant-tabs-tabpane ant-tabs-tabpane-active"
                id="rc-tabs-test-panel-apy"
                role="tabpanel"
                tabindex="0"
              >
                <div
                  aria-busy="true"
                  aria-live="polite"
                  class="ant-spin ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-yp8pcc"
                  style="display: flex; align-items: center; justify-content: center; min-height: 20px; font-size: 14px;"
                >
                  <span
                    class="ant-spin-dot ant-spin-dot-spin"
                  >
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                  </span>
                  <div
                    class="ant-spin-text"
                  >
                    Loading....
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
